### PR TITLE
[PExplicit] Initial architecture with TODOs

### PR DESCRIPTION
**Pull request by aman-goel**
This PR adds the first version of cleaned up explicit-state model checking framework for P, called PExplicit.
This version provides an initial architecture with several remaining TODOs to be completed in future PRs.
There are 2 parts:
- PExplicit IR: This is the new PCompiler backend that generates the IR dedicated for PExplicit.
- PExplicit RT: PExplicit runtime that will run the main explicit-state model checking engine.

Changes include:

[PExplicit IR]:
- Adds the first version of cleaned up basic IR code generator (triggered by hidden mode `--mode explicit`)

[PExplicit RT]:
- Adds PValues classes (like from PObserve) to encode P data types.
- Adds basic harnesses for PMachine, PMonitors, PEvent, etc.
- Adds basic architecture of Scheduler.
- Adds supporting utilities such as logging, exceptions, etc.

### CODE REVIEW

File: Src/PChecker/CheckerCore/CheckerMode.cs [hunk 1]
```diff
@@ -16,5 +16,9 @@ public enum CheckerMode
     /// <summary>
     /// Mode for exhaustive explicit-state search with state-space coverage reporting
     /// </summary>
-    Coverage
+    Coverage,
```
Comment (ankushdesai) [inline Src/PChecker/CheckerCore/CheckerMode.cs:19]: Can we drop coverage?

Comment (aman-goel) [reply Src/PChecker/CheckerCore/CheckerMode.cs:19]: Let's drop this once we have a minimal running version. I wish to retain this for experimental comparison purposes. Note that this is hidden from P CLI.

**Comment by ankushdesai** [comment]
Lets not merge these changes into master. Create a branch, dev/pexplicit_checker and merge the changes into that branch.

File: Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    /**
+     * Status of the run
+     **/
+    @Getter
+    @Setter
+    private static String result = "error";
+    /**
+     * Result of the run
+     **/
+    private final Map<String, StateEvents> allStateEvents = new HashMap<>();    /* Mapping of each machine's state with its corresponding event handlers **/
```
Comment (ankushdesai) [inline Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java:41]: Why do we have this here?

**Review by ankushdesai — CHANGES_REQUESTED** [review]

File: Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java (hunk 2 above)
Comment (aman-goel) [reply Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java:41]: Good catch. This should be put at a different place, inside the State class object.
Will revise and confirm other fields in PExplicitGlobal class. Thanks!

**Review by ankushdesai — APPROVED** [review]

File: Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java (hunk 2 above)
Comment (aman-goel) [reply Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java:41]: To be addressed in next PR.

**Pull request merged**